msal/build.gradle

```diff
@@ -124,6 +133,54 @@ task sourcesJar(type: Jar) {
     destinationDir = reporting.file("$project.buildDir/outputs/jar/")
 }
 
+task createMsalCoverageReportTask() {
+    group = "Msal Reporting"
+    description = "Generate Coverage Report tasks for every build Flavour"
+
+    def buildTypes = android.buildTypes.collect { type ->
+        type.name
+    }
+    def productFlavors = android.productFlavors.collect { flavor ->
+        flavor.name
+    }
+    // When no product flavors defined, use empty string
+    if (!productFlavors) productFlavors.add('')
+    productFlavors.each { productFlavorName ->
+        buildTypes.each { buildTypeName ->
+            def sourceName
+            if (!productFlavorName) {
+                sourceName = "${buildTypeName}"
+            } else {
+                sourceName = "${productFlavorName}${buildTypeName.capitalize()}"
+            }
```

Comment:
see my comments about these in the common PR. same comments apply here

Reply:
added buildVariants to loop over, have a look.